### ISSUE TITLE
Integrate SARATR-X

### ISSUE THREAD

**Issue by lucianchauvin** [Advanced-Vision-and-Learning-Lab/SARIAD#9]
[Paper](https://arxiv.org/abs/2405.09365) [Data/Code](https://github.com/waterdisappear/SARATR-X)

**Comment by lucianchauvin** [comment]
- torch_model can be thought of basically as a feature extractor
- lightning model facilitates the training and stores embeddings/weights

currently trying to implement their `from .SARATRX.classification.model.hivit import HiViT` torch.nn model. Since they have the torch.nn model already done theoretically, we only need to make the lightning model

**Comment by lucianchauvin** [comment]
1. Look into how to create a PyTorch lightning config (see what type of config file Anomalib is looking for) 
2. Simply just integrate the nn.module within the AnomalibModule class
3. Look into how were training just the head.

`SARATR-X/pre_train/main_finetune.py`

**Comment by lucianchauvin** [comment]
<img width="4800" height="1027" alt="Image" src="https://github.com/user-attachments/assets/be722ce9-357f-43a5-9d21-e7f435e0cf95" />

<img width="4784" height="2249" alt="Image" src="https://github.com/user-attachments/assets/6fd0a680-224c-4814-8246-5392745926a2" />

<img width="3743" height="828" alt="Image" src="https://github.com/user-attachments/assets/5657143b-d4e3-43bf-887e-1f22e6824eae" />

**Comment by lucianchauvin** [comment]
<img width="4502" height="884" alt="Image" src="https://github.com/user-attachments/assets/f423a782-1378-44ab-adcc-ad3428c462ce" />

**Comment by lucianchauvin** [comment]
<img width="4787" height="557" alt="Image" src="https://github.com/user-attachments/assets/fb8aeb76-afed-4445-9b85-68e179446016" />

<img width="3531" height="1446" alt="Image" src="https://github.com/user-attachments/assets/7f03c0db-c2ab-4fc5-9548-ae96d179de3b" />

**Comment by lucianchauvin** [comment]
<img width="4097" height="2079" alt="Image" src="https://github.com/user-attachments/assets/62966eb7-d3fb-4f61-bd7e-866876cee74e" />

<img width="3956" height="201" alt="Image" src="https://github.com/user-attachments/assets/bc0af834-b2c9-41b3-aabb-68cc9ba8704f" />

[Seen here](https://github.com/open-edge-platform/anomalib/blob/90e1192dd7d420cb243a93ce17809e43f23fc36d/src/anomalib/data/validators/torch/image.py#L516-L518)

Removing validation gives:

<img width="4790" height="794" alt="Image" src="https://github.com/user-attachments/assets/a7c35b83-a5cf-4a59-99cb-d4ba7ee59f8e" />
and then OOM

HiViTMAE has a in_chans but hard codes this:
<img width="2985" height="337" alt="Image" src="https://github.com/user-attachments/assets/9b258e29-d95a-418b-abac-73b4069618d4" />
I fixed this in their code just to test it out, but then I ran into some other issues. It looks like the image must be greyscale before it hits anything to do with the model.

**Comment by jpeeples67** [comment]
SARATRX now loads the full encoder checkpoint (previously 136 keys were silently randomly initialised) and scores images with the MAE's own multi-scale SAR gradient features. Verified on an RTX 2050 (4 GB) with the real checkpoint: full finetuning at batch size 16 peaks at about 3.4 GiB, and a frozen encoder at about 1.7 GiB, so the reported out-of-memory error does not reproduce at those sizes. It still needs to be trained and evaluated on real MSTAR/HRSID/SSDD data, so this stays open. It was closed automatically when PR #47 was squash-merged.
